packages/notifications/src/utils/validation.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+export enum VALIDATION {
+  SUCCESS = 'success',
+  WARNING = 'warning',
+  ERROR = 'error',
+  INFO = 'info'
+}
+
+export type ValidationTypes = 'success' | 'warning' | 'error' | 'info';
```

Comment:
In many areas you use both `VALIDATION` and `ValidationTypes`. Also, I'm not sure if you have tried to build this yet, but named `ENUMS` probably won't work with our current babel target.

I suggest following the structure of the other packages and having a `type VALIDATION = 'success' | 'warning' | 'error' | 'info'`

Reply:
> I'm not sure if you have tried to build this yet, but named ENUMS probably won't work with our current babel target.

This seems to pass the build, but I had issues with using the named `enum` in development which is why I needed the extra `ValidationTypes`. I will go back and try to consolidate everything to a `type VALIDATION`. Thanks for the suggestion!

Reply:
@hzhu not totally sure it addresses what you're trying to achieve, but consider the types and array constants seen here: https://github.com/zendeskgarden/react-components/blob/next/packages/grid/src/utils/types.ts

Reply:
@austin94 

A named `enum` is [used](https://github.com/zendeskgarden/react-components/blob/next/packages/dropdowns/src/styled/field/StyledMessage.ts#L15) in the `react-dropdowns`. Could you confirm if named `enums` are okay to use? I'm using a named `enum` in this PR, and it seems to work. 

Also, I didn't find any packages in the `next` branch having `type VALIDATION = 'success' | 'warning' | 'error' | 'info'`. Which packages can I find that?

@jzempel 

That file is beneficial, thank you for sharing! I went ahead and used it as a guideline. Instead of using creating an `Array` of types (like [this](https://github.com/zendeskgarden/react-components/blob/next/packages/grid/src/utils/types.ts#L9)), I created an `enum` of types because `react-notification` uses object property dot notation [here](https://github.com/zendeskgarden/react-components/blob/next/packages/notifications/src/Alert.js#L34) and [here](https://github.com/zendeskgarden/react-components/blob/next/packages/notifications/src/Notification.js#L34) — a behavior that a typed array would not support.

I ended up with the following, which worked well for me:

```ts
/*
./react-components/packages/notifications/src/utils/types.ts
*/

export type VALIDATION_TYPES = 'success' | 'warning' | 'error' | 'info';

export enum VALIDATION {
  SUCCESS = 'success',
  WARNING = 'warning',
  ERROR = 'error',
  INFO = 'info'
}
```

Reply:
I would avoid using that value, even if it's in use with `react-dropdowns`. That was the first TS package so there are some things in there we need to align with our other packages.

> created an enum of types because react-notification uses object property dot notation here and here

This dot-notation is a side-effect of using a CONSTANTS object. With TS this isn't needed. We can rely on the type-system to enforce this. You can just use `validation === 'success'` instead of `validation === VALIDATION.SUCCESS`

Reply:
That sounds good. The code has been updated to not use property lookup on an object. That way it now relies on the type-system alone to enforce safety.